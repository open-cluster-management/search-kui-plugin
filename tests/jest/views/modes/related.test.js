/*******************************************************************************
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
// Copyright (c) 2021 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'
const related = require('../../../../dist/views/modes/related')

const items = [
  {kind: require('../../../data/configmaps')}
]
const configmap = items[0].kind.configmap

describe('Resources/Query sidecar', () => {
  // Resource's Related Resources tab
  describe('related mode', () => {
    const spy = jest.spyOn(related, 'relatedTab')
    const _ = related.relatedTab(configmap)

    it('should render a tab entity for the selected resource related resources', () => {
      expect(spy).toBeCalled()
      expect(related.relatedTab(configmap)).toMatchSnapshot()
    })
  })

  describe('build related', () => {
    const spy = jest.spyOn(related,  'buildRelated')
    let _  = related.buildRelated(configmap)

    it('should build a related resources node element - structured list', () => {
      expect(spy).toBeCalled()
      expect(related.buildRelated(configmap)).toMatchSnapshot()
    })

    _  = related.buildRelated(configmap, 'query')

    it('should build a related resources node element - clickable list', () => {
      expect(spy).toBeCalled()
      expect(related.buildRelated(configmap, 'query')).toMatchSnapshot()
    })
  })
})
