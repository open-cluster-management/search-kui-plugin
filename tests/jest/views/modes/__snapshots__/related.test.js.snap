// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources/Query sidecar build related should build a related resources node element - clickable list 1`] = `
<div
  class="scrollable bx--tile-related"
>
  <a
    class="bx--link bx--tile bx--tile--clickable"
    tabindex="0"
  >
    <div
      class="bx--tile-container"
    >
      <span
        class="bx--structured-list-td-related-header"
      >
        1
      </span>
      <br />
      <span
        class="bx--structured-list-td-body"
      >
        Related release
      </span>
    </div>
  </a>
</div>
`;

exports[`Resources/Query sidecar build related should build a related resources node element - structured list 1`] = `
<div
  class="scrollable bx--structured-list--summary"
>
  <section
    aria-label="Structured list section"
    class="bx--structured-list"
  >
    <div
      class="bx--structured-list-tbody"
    >
      <div
        class="bx--structured-list-row bx--structured-list-rowclick"
      >
        <div
          class="bx--structured-list-td"
          tabindex="0"
        >
          <span
            class="bx--structured-list-td-related-header"
          >
            1
          </span>
          <br />
          <span
            class="bx--structured-list-td-body"
          >
            Related release
          </span>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Resources/Query sidecar related mode should render a tab entity for the selected resource related resources 1`] = `
Object {
  "badges": Array [
    "audit-logging-fluentd-ds-elk-config",
  ],
  "content": <div
    class="scrollable bx--structured-list--summary"
  >
    <section
      aria-label="Structured list section"
      class="bx--structured-list"
    >
      <div
        class="bx--structured-list-tbody"
      >
        <div
          class="bx--structured-list-row bx--structured-list-rowclick"
        >
          <div
            class="bx--structured-list-td"
            tabindex="0"
          >
            <span
              class="bx--structured-list-td-related-header"
            >
              1
            </span>
            <br />
            <span
              class="bx--structured-list-td-body"
            >
              Related release
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "isEntity": true,
  "modes": Array [
    Object {
      "defaultMode": true,
      "direct": [Function],
      "label": "Related Resources",
      "leaveBottomStripeAlone": true,
      "mode": "related",
    },
  ],
  "name": "audit-logging-fluentd-ds-elk-config",
  "packageName": "kube-system",
  "type": "custom",
  "viewName": "configmap",
}
`;
